src/docs/sphinx/dev_guide/gpu_porting.rst

```diff
@@ -0,0 +1,302 @@
+RAJA::kernel
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+``RAJA::kernel`` is for more complex and/or nested loops.
+
+This is used infrequently, mainly seen only in a `few unit tests`_.
+
+Your general go-to will be ``axom::for_all``.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Useful Links
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+`RAJA Loops`_ - Covers for_all, kernel, and other loops
+
+
+.. _gpu-execution-spaces-label:
+
+================================
+Execution Spaces & Policies
+================================
+
+Axom's execution spaces can be found in
+`axom/core/execution/execution_space.hpp <https://github.com/LLNL/axom/blob/develop/src/axom/core/execution/execution_space.hpp>`_
+
+Axom's execution spaces are derived from an execution_space traits class,
+containing RAJA execution policies and default Umpire memory allocators
+associated with each space.
+
+Axom currently supports `4 execution spaces`_:
+
+* ``SEQ_EXEC`` - Sequential execution policies on host
+* ``OMP_EXEC`` -  OpenMP execution policies on host
+* ``CUDA_EXEC`` - CUDA execution policies in Unified Memory (host + device)
+* ``HIP_EXEC`` - HIP execution policies in Unified Memory (host + device)
+
+Additionally, ``HIP_EXEC`` and ``CUDA_EXEC`` are templated by the
+number of threads and SYNCHRONOUS or ASYNC execution.
+
+.. literalinclude:: ../../../axom/core/execution/internal/cuda_exec.hpp
+   :start-after:  _cuda_exec_start
+   :end-before:  _cuda_exec_end
+   :language: C++
+
+Each execution space (``axom::execution_space<ExecSpace>``) provides:
+
+* Axom policies that are type aliases of RAJA policies to be used with kernels,
+  RAJA types, and RAJA operations
+
+  * ``loop_policy`` - For `RAJA scans`_ and other operations; ``axom::for_all``
+    uses the loop_policy from the templated execution space.
+  * ``reduce_policy`` - For  `RAJA reduction types`_
+  * ``atomic_policy`` - For `RAJA atomic operations`_
+  * ``sync_policy`` - For Axom's `synchronize`_ function, which is a wrapper
+    around ``RAJA::synchronize()``
+
+* Umpire allocator defaults
+
+  * ``memory_space`` - The memory space abstraction for use by
+    :ref:`Core Containers<core-containers>` like ``axom::Array``.
+  * ``allocatorID()`` - Gets the allocator ID for the Umpire resource to use in
+    this execution space.
+
+* General information on the execution space
+
+  * ``name()`` - Name of the execution space
+  * ``onDevice()`` - Is the execution space on device? (True/False)
+  * ``valid()`` - Is the execution space valid? (True)
+  * ``async()`` - Is the execution space asynchronous? (True/False)
+
+The mint component also provides a set of nested execution policies
+located at
+`axom/mint/execution/internal/structured_exec.hpp <https://github.com/LLNL/axom/blob/develop/src/axom/mint/execution/internal/structured_exec.hpp>`_
+to be used with
+``RAJA::kernel`` e.g. for iterating over mint meshes.
+
+.. note::
+
+  When Axom is built without RAJA, only ``SEQ_EXEC`` is available
```

Comment:
This note is a bit confusing. It's not clear what Umpire has to do with available execution policies since Umpire is not used to execute anything, only to manage memory.